lib/mongo/collection/view.rb

```diff
@@ -132,7 +132,7 @@ def hash
       def initialize(collection, filter = {}, options = {})
         validate_doc!(filter)
         @collection = collection
-        parse_parameters!(BSON::Document.new(filter.freeze), BSON::Document.new(options.freeze))
+        parse_parameters!(BSON::Document.new(filter.dup.freeze), BSON::Document.new(options.dup.freeze))
```

Comment:
Oh for sure. I didn't look that deep but yes, certainly down lower in the code would be great.